snapcraft/tests/test_sources.py

```diff
@@ -657,27 +657,51 @@ def test_init_with_source_depth_raises_exception(self):
 
 class TestLocal(tests.TestCase):
 
-    def test_pull_with_source_a_parent_of_current_dir(self):
+    def test_pull_with_source_the_parent_of_current_dir(self):
         snapcraft_files_before_pull = copy.copy(common.SNAPCRAFT_FILES)
 
         # Verify that the snapcraft root dir does not get copied into itself.
         os.makedirs('subdir')
 
-        cwd = os.getcwd()
-        os.chdir('subdir')
-        local = sources.Local('..', 'foo')
-        local.pull()
-        os.chdir(cwd)
-
-        self.assertTrue(
-            'subdir' not in os.listdir(os.path.join('subdir', 'foo')))
+        for snap_dir in [f for f in common.SNAPCRAFT_FILES if '.' not in f]:
+            cwd = os.getcwd()
+            os.chdir('subdir')
+            source_dir = os.path.join(snap_dir, 'foo')
+            local = sources.Local('..', source_dir)
+            local.pull()
+            os.chdir(cwd)
```

Comment:
This is inside the loop. But addCleanup would run after the test finishes. You can use test scenarios instead of a for inside the test, and then IMO things are clearer.
And I'm not sure if I'm missing something, but tests.TestCase is already using TempCWD, so you never have to worry about leaving the cwd in a different dir, the parent test will clean that up.